packages/@react-aria/listbox/src/useOption.ts

```diff
@@ -121,7 +121,9 @@ export function useOption<T>(props: AriaOptionProps, state: ListState<T>, ref: R
   if (isVirtualized) {
     let index = Number(item?.index);
     optionProps['aria-posinset'] = Number.isNaN(index) ? undefined : index + 1;
-    optionProps['aria-setsize'] = getItemCount(state.collection);
+    // TODO: this is not great, but the loader sentinel is always in the collection even when loading isn't currently in progress.
+    // This same issue applies to the other collection elements, namely the row counts calculated for the top level parent element
+    optionProps['aria-setsize'] = getItemCount(state.collection, (node) => node.type === 'item' || (node.type === 'loader' && node.props.isLoading));
```

Comment:
these are kinda weird in RAC/S2 when there are sections, the `setsize` is still based on the whole collection but the `posinset` is based off the index within the section since the collection calculated index is different in new collections